src/s1reader/s1_etad.py

```diff
@@ -0,0 +1,202 @@
+"""Wrapper of s1etad module to read S1-ETAD products."""
+import datetime
+import glob
+import os
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy.interpolate import RectBivariateSpline
+
+try:
+    import s1etad
+except ImportError:
+    raise ImportError('Can NOT import s1etad (https://gitlab.com/s1-etad/s1-etad)!')
+
+
+
+def get_eta_correction_from_slc_burst(slc_burst, eta_dir, corr_type='sum', include_tropo=True,
+                                      resample=True, plot=False, verbose=True, unit='second'):
+    """Get the ETAD correction for one S1 burst.
+
+    Parameters
+    ----------
+    slc_burst: Sentinel1BurstSlc
+        Sentinel-1 burst object from sentinel1_reader
+    eta_dir: str
+        Sentinel-1 ETAD product directory
+    corr_type: (list of) str
+        Sentinel-1 ETAD correction type:
+        sar, atm, sum, bistatic, doppler, fmrate, geodetic, ionospheric, tropospheric
+        where sar = bistatic + doppler + fmrate
+              atm = ionospheric + tropospheric
+              sum = sar + atm + geodetic
+    resample: bool
+        resample the low resolution ETA to the full resample SLC size
+    unit: str
+        output ETA correction unit:
+        pixel, second (meter is not recommended, as time info is more consistent and universal in SAR)
+
+    Returns:
+    ----------
+    slc_rg_corr: np.ndarray in float32 in size of (lines, samples)
+        S1 ETAD correction in range   direction in the unit of meter or second
+    slc_az_corr: np.ndarray in float32 in size of (lines, samples)
+        S1 ETAD correction in azimuth direction in the unit of meter or second
+    """
+    vprint = print if verbose else lambda *args, **kwargs: None
+    if unit not in ['second', 'meter', 'pixel']:
+        raise ValueError(f'Un-recognized input unit={unit}!')
+    meter = (unit == 'meter')
```

Comment:
I am wondering if we should add a warning when the user selects `meter` as a unit. This warning is hidden in the doc string but probably it is more visible to an unexperienced user with a print message

Reply:
We recommend using second because it's more accurate than meter for SAR data record, but I don't have a number on how big of a difference it could be. Is this the reason you wanted a warning msg?

Reply:
Yeah! I think an unexperienced user won't think about it. So, if we throw a warning, we might get its attention to that. On the other hand, this might be useless if we do not know what the discrepancy in accuracy between meters and seconds XD